counsel.el

```diff
@@ -6379,7 +6385,44 @@ sub-directories that builds may be invoked in."
   (let ((srcdir (counsel--compile-root)))
     (when (directory-files (or blddir srcdir) nil
                            counsel-compile-make-pattern t)
-      (counsel--compile-get-make-targets srcdir blddir))))
+      (counsel--compile-get-make-targets
+       'counsel-compile--probe-make-targets srcdir blddir))))
+
+
+(defun counsel-compile--probe-make-help (dir)
+  "Return a list of Make targets based on help for DIR.
+
+It is quite common for a 'make help' invocation to return a human
+readable list of targets.  Often common targets are marked with a
+leading asterisk.  The exact search pattern is controlled by
+`counsel-compile-help-pattern'."
+  (let ((default-directory dir)
+        (primary-targets nil)
+        (targets nil))
+    ;; only proceed if the help target exists
+    (when (= (call-process "make" nil t nil "-q" "help") 1)
+      (with-temp-buffer
+        (when (= (call-process "make" nil t nil "help") 0)
+          (goto-char (point-min))
+          (while (re-search-forward counsel-compile-help-pattern nil t)
+            (let ((text (match-string-no-properties 2)))
+            (if (match-beginning 1)
```

Comment:
Indentation seems off here. See also my previous comment about the value of `(match-beginning 1)` here.